apps/dashboard/app/models/hpc_module.rb

```diff
@@ -39,4 +45,12 @@ def to_s
   def default?
     version.nil?
   end
+
+  def ==(other)
```

Comment:
This is clearly ruby magic.  It is also very cryptic.  Can you please add a few comments describing what & why.

Reply:
> It is redefining the comparison operator `==` for this class to ensure that the string representation of objects is what is being compared.

Great explanation. That should go into a comment above the definition. ;)

Reply:
The pattern is very common in ruby code within a class to redefine comparison for the class, so it is one of those things that being familiar with is somewhat necessary. That said, i have an aversion to comments in ruby as the code should be expressive itself without the need for comments. Though I don't mind comments around expected return types.

You can see other examples of redifining `==` in `app/ood_app.rb` and `models/batch_connect.rb` that I know of.

Reply:
If we redefine out of the box functionality, then we should comment that redefinition so when the next guy/gal looks at it they know why it is being done and not to undo it.

Reply:
comments are normal things in software development. not only are they a normal practice, they are actually recommended.

Reply:
I only needed it in once place - so for simplicity I just removed it altogether

https://github.com/OSC/ondemand/pull/1930/commits/2dec64f7634dd68ee5d8d72c665e6e08845ad9d4